File: algorithms/Set Matrix Zeroes/Set Matrix Zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int rows = matrix.size(), cols = matrix[0].size();
        int rowFlag = 0, colFlag = 0;
        
        for(int i = 0; i < rows; i++){
            if(matrix[i][0] == 0){
                rowFlag = 1;
                break;
            }
        }
        
        for(int i = 0; i < cols; i++){
            if(matrix[0][i] == 0){
                colFlag = 1;
                break;
            }
        }
        
        for(int i = 0; i < rows; i++){
            for(int j = 0; j < cols; j++){
                if(matrix[i][j] == 0){
                    matrix[i][0] = 0;
                    matrix[0][j] = 0;
                }
            }
        }
        
        for(int i = 1; i < rows; i++){
            for(int j = 1; j < cols; j++){
                if(matrix[i][0] == 0 || matrix[0][j] == 0){
                    matrix[i][j] = 0;
                }
            }
        }
        
        if(rowFlag){
            for(int i = 0; i < rows; i++){
                matrix[i][0] = 0;
            }
        }
        
        if(colFlag){
            for(int i = 0; i < cols; i++){
                matrix[0][i] = 0;
            }
        }
    }
};

File: scripts/algorithms/S/Set Matrix Zeroes/Set Matrix Zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        rows = len(matrix)
        cols = len(matrix[0])
        visited=set()
        for r in range(rows):
            for c in range(cols):
                
                if matrix[r][c]==0 and (r,c) not in visited :
                    for t in range(cols):
                        if matrix[r][t]!=0:
                            matrix[r][t]=0
                            visited.add((r,t))
                    for h in range(rows):
                        if matrix[h][c]!=0:
                            matrix[h][c]=0
                            visited.add((h,c))
        ##Time Complexity :- O(m*n)
        ##Space Complexity:- O(m+n)
		
		```